### DISCUSSION TITLE
Run highlighted code only

### DISCUSSION THREAD

**Discussion by ricardovfreixo** [tconbeer/harlequin#500] [Ideas (Feature Requests)]
I've been using dbeaver for a long time now. The one feature that brings me back is the possibility of saving huge scripts with many snippets, highlighting the snippet I want to run and doing a ctrl-enter to run just the highlighted bit. I usually have scripts for everyday tasks with sequential snippets and many notes. Maybe this is just a personal thing. If this is something more people want to see, I can even work on it. Do let me know.

**Comment by tconbeer** [comment]
Harlequin already works this way, unless the adapter supports validation and the highlighted snippet isn't a valid sql query (this validation is only in duckdb and bigquery afaik). If that is not the behavior you're seeing, could you please open a bug report?

**Comment by ricardovfreixo** [comment]
I will install other adapter and will test out. maybe is just a bug then. I am using the postgres adapter. and doesn't work like that. ctrol-enter insert another line instead of running. That is fine with me because ctrl-J works fine. but may be either from the adapter or my terminal (I am using terminator). I will run some tests with other terminal and other adapter and will let you know

**Reply by tconbeer** [reply]
It sounds like you need to configure your terminal to forward the ctrl+enter key press to the shell. Seems like it's intercepting that key and sending a different control code.
